File: jupyter2/problems/parametric/function_in_parameters/ml/function_in_parameters.ml
(** begin my answer *)

(** end my answer *)

let main () =
  assert (app1 f = 1.0);
  assert (app1 g = [1.0]);
  assert (app  f 4 = 4);
  assert (app  h ["bye"] = Some "bye");
  Printf.printf "OK\n"
;;

main()

File: jupyter2/problems/parametric/function_in_parameters/go/function_in_parameters.go
package main
import (
	"fmt"
	"slices"
)

/** begin my answer */

/** end my answer */
func main() {
	if !(app1(f) == 1.0)                                       { panic("wrong") }
	if !(slices.Equal(app1(g), []float64{1.0}))                { panic("wrong") }
	if !(app(f, 4) == 4)                                       { panic("wrong") }
	if r := app(h, []string{"bye"}); !(r.ok && r.val == "bye") { panic("wrong") }
	fmt.Println("OK")
}
